restructure code base

mly32 opened the pull request:
### Summary
<!-- What changes were made? What features were added? What bugs were fixed? -->
This PR addresses Jira ticket CISLUNAR-80.

Added pre-commit hooks, and updated setup.py and requirements.

Pre-commit hooks for formatting and linting are currently turned off, in fear of causing thousands of lines of code changes.

The requirements in setup.py need to be updated because it was not clear what goes in "rpi" and what goes in normal install requirements.

### Notes

Please see the new CONTRIBUTING.md for instructions.

### Todo

Citing, and also adding licensing.

tmf97 commented on setup.py:
```diff
@@ -16,53 +12,72 @@
     "busio",
-    "adafruit-circuitpython-bno055",
+    "picamera",
+    "RPi.GPIO",
```
We don't actually use the `RPi.GPIO` library anywhere in our flight software. All of the pi's pin functionality is controlled using the `pigpio` module

tmf97 commented on requirements_pi.txt:
```diff
@@ -1,18 +0,0 @@
-numpy-quaternion
-astropy
-adafruit-circuitpython-ds3231
-bitstring
```
Our flight computer setup procedure depended on this file. Can you please also update the "Flight Software Setup" section of https://cornell.app.box.com/file/765928258516

tmf97 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,155 @@
+
+- Explain in detail how it would work.
+- Keep the scope as narrow as possible, to make it easier to implement.
+- Remember that this is a volunteer-driven project, and that contributions are
```
```suggestion
- Remember that this is a student volunteer-driven project, and that contributions are
```

Just being pedantic🤪

tmf97 commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,17 @@
+repos:
```
what is pre-commit and how do I use it? Looks very cool from a quick google search

tmf97 reviewed (changes requested):
I'm a big fan of this! Just some minor things we need to tidy up then we're good to go

tmf97 commented on README.md:
```diff
@@ -1,74 +1,37 @@
 # Cislunar Explorers Flight Software
 
+[![GitHub Workflow Status](https://img.shields.io/github/workflow/status/Cislunar-Explorers/FlightSoftware/Pytest%20testing)](https://github.com/Cislunar-Explorers/FlightSoftware/actions)
```
Adding a build passing badge gives us a false sense of security - I think our code testing coverage is less than 15%. So, worst case scenario, 85% of our code could be broken and our build would still be passing. Adding a code coverage badge would be a good indicator of how sparse our automated testing actually is: https://stackoverflow.com/questions/29295965/python-coverage-badges-how-to-get-them

Of course, we can't assign a badge to the coverage provided by all of our HITL (Hardware In The Loop) tests: https://cornell.app.box.com/file/734055246538, however, HITL tests quickly become invalid as code gets iterated so there is no feasible way to easily quantify our coverage.

tmf97 replied:
Also, pytest doesn't run any of our Opnav tests, all of which are failing

mly32 commented:
Good points. I agree with the point about the badge and updated the readme.

I also changed the box setup documentation but would like to have that looked over.

By the way, I've purposely changed the venv name from `cislunar-ven` to `venv` to make it clear that this is a new environment and also because `venv` is pretty standard. Hope that doesn't cause any problems.

After running `pre-commit install` as described in contributing, pre-commit will add relevant pre-commit hooks into the `.git` folder. Essentially, before anyone makes a commit, these scripts are run on the project. They can be set up for lining and automatic formatting on only the changed files (this makes the checks faster). This way, the code on master is always consistent. Tests can also be added but it seems redundant due to the CI already in place.

(to see it in action, change a file, run `git add .`, then try `git commit -m "commit message"`)

Again, the lining and formatting hooks are turned off right now to prevent major code changed. Right now, there are some basic fixed pre-commit provides in place, such as `end-of-file-fixer`.

tmf97 commented on setup.py:
```diff
@@ -42,22 +42,12 @@
-    "pre-commit",
-    "watchdog"
-]
+DEV_REQUIRES = ["black", "flake8", "pip", "pre-commit", "watchdog"]
```
What is watchdog?

mly32 replied:
watchdog is a file monitoring utility. It comes with a shell command that can be used to track changes in documentation and live update documentation (think of nodemon for nodejs or something).

Its currently not doing anything because I'm still having trouble getting auto documentation with Sphinx working. Its because lots of examples I see online have this folder structure which our project doesn't use:

- root
  - docs
  - project_name
    - submodules
  - tests
  - ...

I'm still trying to figure out how to get docs working. If restructuring the code to the format above works, I would do it but I fear its going to change too much.

tmf97 reviewed (approved):
Solid stuff!

mly32 commented:
@tmf97 yah, no problem. I was still trying to solve the docs issue, but I'll merge this in first.

tmf97 commented:
All good. In general we should try to keep PRs as small and nimble as possible. The docs issue can be it's own PR, even if there's only one jira ticket for both
